pkg/backup/backup/backup_cleaner.go

```diff
@@ -52,7 +53,73 @@ func NewBackupCleaner(deps *controller.Dependencies, statusUpdater controller.Ba
 }
 
 func (bc *backupCleaner) Clean(backup *v1alpha1.Backup) error {
-	if backup.DeletionTimestamp == nil || !v1alpha1.IsCleanCandidate(backup) || v1alpha1.NeedNotClean(backup) {
+	if backup.DeletionTimestamp == nil {
+		// The backup object has not been deleted，do nothing
+		return nil
+	}
+
+	if err := bc.CleanLogBackup(backup); err != nil {
+		return err
+	}
+	klog.Infof("start to clean backup %s/%s", backup.GetNamespace(), backup.GetName())
+	return bc.CleanData(backup)
+}
+
+func (bc *backupCleaner) CleanLogBackup(backup *v1alpha1.Backup) error {
+	if backup.Spec.Mode != v1alpha1.BackupModeLog {
+		return nil
+	}
+	if backup.Spec.BR == nil {
+		return fmt.Errorf("backup %s/%s spec.BR shouldn't be nil", backup.GetNamespace(), backup.GetName())
+	}
+	if !v1alpha1.IsLogBackupAlreadyStart(backup) {
+		return nil
+	}
+	if v1alpha1.IsLogBackupAlreadyStop(backup) {
+		return nil
+	}
+
+	ns := backup.GetNamespace()
+	name := backup.GetName()
+	stopLogJobName := backup.GetCleanLogBackupJobName()
+
+	var err error
+	_, err = bc.deps.JobLister.Jobs(ns).Get(stopLogJobName)
+	if err == nil {
+		// already have a clean job running，return directly
+		return nil
+	}
```

Comment:
do we need to differ the NotFound error with other errors?

Reply:
Yes